products.js

```diff
@@ -253,56 +250,56 @@ class DiffStats {
+  let nodesMap = new Map();
+  const numberToCss = JSON.parse(readFileSync(join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', item, 'nodes.txt'), 'utf-8'));
+  const elems = sourceDom.getElementsByTagName('*');
+  console.log(sourceDom.getElementsByTagName('*').length, Object.keys(numberToCss).length);
 
   //Match each node from the sourceDom to its css, and store in the global dict
-  for(let i = 0; i < Object.keys(nodesDict).length; i++){
-    currDict[elems[i]] = nodesDict[i.toString()];
+  for(let i = 0; i < elems.length; i++){
+    nodesMap.set(elems[i], numberToCss[i]);
   }
+  console.log(nodesMap.size);
+  return nodesMap;
 }
 
-
+/*
+ * Calculate overall score for one feature
+ * Args: folders - list of all test folders; feature - title/image/price; coeffs - tuning coeffs if any
+ */
 function deviationScore(folders, feature, coeffs = []) {
-    var stats;
-    if(feature === 'image') {
-      stats = new DiffStats(tunedImageFnodes(...coeffs), feature);
-    } else if (feature === 'title') {
-      stats = new DiffStats(tunedTitleFnodes(...coeffs), feature);
-    } else {
-      stats = new DiffStats(tunedPriceFnodes(...coeffs), feature);
-    }
+    const stats = new DiffStats(tuningRoutines[feature], feature);
 
     //For each test file, create the object-> css map, and run the comparison function
-    folders.forEach(function(item){
-      if(['macys', 'swatch'].includes(item)){return;} //random jsdom script tag parsing errors, ignore for now
-      const domFromFile = fileName => staticDom(readFileSync(join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', item, fileName)));
-
-      createDict(item, domFromFile('source.html'));
-      var pair = [domFromFile('expected-' + feature + '.html'), domFromFile('source.html')];
-      stats.compare(...pair);
-      console.log(stats.score());
-      currDict = {};
+    folders.forEach(function(store){
+      if(['macys', 'swatch'].includes(store)){return;} // HACK: random jsdom script tag parsing errors, ignore for now
+      const domFromFile = fileName => staticDom(readFileSync(join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', store, fileName)));
+      const dict = createDict(store, domFromFile('source.html'));
+      const pair = [domFromFile('expected-' + feature + '.html'), domFromFile('source.html')];
+      // stats.compare(pair[0], pair[1], dict);
+      // console.log(stats.score());
     });
 
-    return stats.score();
+    // return stats.score();
 }
 
 if (require.main === module) {
-    const features = ['title', 'image', 'price'];
-    const dirs = p => readdirSync(p).filter(f => statSync(p+"/"+f).isDirectory());
-    const folders = dirs(join(dirname(__dirname), 'fathom-products' , 'product_classification_test_data'));
-    // const folders = ['amazon'];
+    const folders_in = p => readdirSync(p).filter(f => statSync(p + "/" + f).isDirectory());
```

Comment:
I gave you bad spelling advice (which I later corrected but too late). In JS we do camelCaseCapitalization. underscore_naming is for Python.